libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -655,48 +703,19 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
                         )}
                     </MediaQuery>
                 }
-                body={
-                    isAllFiltered ? (
-                        <MediaQuery query={MediaQueries.IS_MOBILE_DEVICE}>
-                            {(isMobile) => (
-                                <AttributeDropdownAllFilteredOutBody
-                                    parentFilterTitles={parentFilterTitles}
-                                    onApplyButtonClick={onApplyButtonClicked}
-                                    onCancelButtonClick={onCloseButtonClicked}
-                                    isMobile={isMobile}
-                                />
-                            )}
-                        </MediaQuery>
-                    ) : hasNoData ? (
-                        <NoData noDataLabel={props.intl.formatMessage({ id: "attributesDropdown.noData" })} />
-                    ) : (
-                        <AttributeDropdownBody
-                            items={state.validOptions?.items ?? []}
-                            totalCount={totalCount ?? LIMIT}
-                            selectedItems={state.selectedFilterOptions}
-                            isInverted={state.isInverted}
-                            isLoading={
-                                (!state.validOptions?.items && isElementsLoading()) ||
-                                isTotalCountLoading() ||
-                                isParentFilterTitlesLoading()
-                            }
-                            searchString={state.searchString}
-                            onSearch={onSearch}
-                            onSelect={onSelect}
-                            onRangeChange={onRangeChange}
-                            onApplyButtonClicked={onApplyButtonClicked}
-                            onCloseButtonClicked={onCloseButtonClicked}
-                            applyDisabled={getNumberOfSelectedItems() === 0}
-                            parentFilterTitles={parentFilterTitles}
-                            showItemsFilteredMessage={showItemsFilteredMessage(
-                                isElementsLoading(),
-                                resolvedParentFilters,
-                            )}
-                        />
-                    )
-                }
                 ref={dropdownRef}
                 onOpenStateChanged={onDropdownOpenStateChanged}
+                body={
+                    props.renderBody
+                        ? props.renderBody({
+                              ...bodyProps,
```

Comment:
You should be able to make these "extended" props and pass them both to the `props.renderBody` and `renderDefaultBody`. But the current solution is fine too.

Reply:
I have made all extended props optional so we can use them on renderDefaultBody as well. Let me know if the solution is fine with you.